lib/src/core/services/dataloader_service.dart

```diff
@@ -0,0 +1,43 @@
+import "package:kana_to_kanji/src/core/dataloaders/group_dataloader.dart";
+import "package:kana_to_kanji/src/core/dataloaders/kana_dataloader.dart";
+import "package:kana_to_kanji/src/core/dataloaders/kanji_dataloader.dart";
+import "package:kana_to_kanji/src/core/dataloaders/vocabulary_dataloader.dart";
+import "package:kana_to_kanji/src/core/services/cleanup_service.dart";
+import "package:kana_to_kanji/src/core/services/sync_service.dart";
+import "package:kana_to_kanji/src/locator.dart";
+
+class DataloaderService {
+  final SyncService _syncService = locator<SyncService>();
+  final GroupDataLoader _groupDataLoader = locator<GroupDataLoader>();
+  final KanaDataLoader _kanaDataLoader = locator<KanaDataLoader>();
+  final KanjiDataLoader _kanjiDataLoader = locator<KanjiDataLoader>();
+  final VocabularyDataLoader _vocabularyDataLoader =
+      locator<VocabularyDataLoader>();
+  final CleanUpService _cleanUpService = locator<CleanUpService>();
+
+  /// Load all the static data from the API. But to optimise the calls,
+  /// a first one is done to `GET /sync` to know which data must be updated.
+  Future<void> loadStaticData() async {
```

Comment:
Do we want to extend this service to support the User also?

Reply:
Done